File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private int result=0;
    
    private void helper(TreeNode current,int total){
        if(current.left==null && current.right==null){
            result+=total;
            return;
        }
        
        if(current.left!=null) {
            helper(current.left,10*total+current.left.val);
        }
        
        if(current.right!=null){
            helper(current.right,10*total+current.right.val);
        }
    }
    public int sumNumbers(TreeNode root) {
        if(root==null) return 0;
        
        helper(root,root.val);  
        return result;
    }
}

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.js
var sumNumbers = function (root) {
    let total = 0;

    let dfs = (node, sum) => {
        if (!node.left && !node.right) total += sum * 10 + node.val;
        else { 
            if (node.left) dfs(node.left, sum * 10 + node.val);
            if (node.right) dfs(node.right, sum * 10 + node.val);
        }
    }
    
    dfs(root, 0);
    return total;
};